src/common/controllers/healthController.js

```diff
@@ -1,17 +1,17 @@
 'use strict';
 
 const database = require('../../database/database');
+const streamingManager = require('../../streaming/manager');
 
-module.exports.check = function (req, res, next) {
-    const errors = {};
-    database.ping()
-        .catch((error) => {
-            errors.database = error && error.message ? error.message : error;
-        }).then(() => {
-            if (Object.keys(errors).length > 0) {
-                return res.status(503).json({ status: 'DOWN', errors: errors });
-            } else {
-                return res.json({ status: 'OK' });
-            }
-        });
+module.exports.check = async function (req, res, next) {
+    const errors = await Promise.all([
+        database.ping(),
+        streamingManager.health()
+    ]);
+    const filteredErrors = errors.filter((element) => !!element);
```

Comment:
I wouldn't set it down if there is an issue with streamingManager

Reply:
streaming manager doesn't throw an error on the health check so it won't report as down

Reply:
done

Reply:
i suggest to do check the health of it and also report in the endpoint response
just not set the status to 5xx if there is a problem

currently, it doesn't await the response so I think it's there to 'reheal'

one more point when checking Kafka health is to limit it by a timeout of 1-2 second otherwise if the response is long k8s health check will get timeout and mark itself as down

Reply:
good point, done